On Solaris/illumos, sun is the preprocessor define that identifies us.
Rename to anything else.

--- a/xs_unix_socket.h	Thu Aug  8 05:35:24 2024
+++ b/xs_unix_socket.h	Mon Aug 12 14:44:29 2024
@@ -20,15 +20,15 @@
     int rs = -1;
 
     if ((rs = socket(AF_UNIX, SOCK_STREAM, 0)) != -1) {
-        struct sockaddr_un sun = {0};
+        struct sockaddr_un snac_sun = {0};
         mode_t mode = 0666;
 
-        sun.sun_family = AF_UNIX;
-        strncpy(sun.sun_path, path, sizeof(sun.sun_path));
+        snac_sun.sun_family = AF_UNIX;
+        strncpy(snac_sun.sun_path, path, sizeof(snac_sun.sun_path));
 
         unlink(path);
 
-        if (bind(rs, (struct sockaddr *)&sun, sizeof(sun)) == -1) {
+        if (bind(rs, (struct sockaddr *)&snac_sun, sizeof(snac_sun)) == -1) {
             close(rs);
             return -1;
         }
@@ -58,12 +58,12 @@
     int d = -1;
 
     if ((d = socket(AF_UNIX, SOCK_STREAM, 0)) != -1) {
-        struct sockaddr_un sun = {0};
+        struct sockaddr_un snac_sun = {0};
 
-        sun.sun_family = AF_UNIX;
-        strncpy(sun.sun_path, path, sizeof(sun.sun_path));
+        snac_sun.sun_family = AF_UNIX;
+        strncpy(snac_sun.sun_path, path, sizeof(snac_sun.sun_path));
 
-        if (connect(d, (struct sockaddr *)&sun, sizeof(sun)) == -1) {
+        if (connect(d, (struct sockaddr *)&snac_sun, sizeof(snac_sun)) == -1) {
             close(d);
             d = -1;
         }
